README.md

```diff
@@ -0,0 +1,321 @@
+
+##### Why are there so many factory interfaces in Captain?
+
+All of the default waypoint implementations in captain offer two ways of creating a waypoint:
+1. Using factories (`RequestSubmitterFactory`, `HandlePersistorFactory`, `StatusRetrieverFactory`, etc)
+2. Not using factories (`RequestSubmitter`, `HandlePersistor`, `StatusRetriever`, etc)
+
+Each request that gets processed by captain will call the `create` method of the corresponding manifest factory. That means that all the resources that are used to call `create` will be used by each of those requests. That can be problematic. Let's say one of your submitter looks as follows:
+
+```
+DummySubmitter(ServiceClient1 serviceClient1, ServiceClient2 serviceClient2) {
+this.serviceClient1 = serviceClient1;
+this.serviceClient2 = serviceClient2;
+}
+```
+Whenever you initialize that submitter, you need to already have the service clients initialized (let's assume these clients eagerly connect and have some sort of persistent connection), meaning that you'll already have made a connection to that service. Let's say that you've decided to not use factories. In addition, you have 10 steps in your manifest and all of them make connections to different services. That means that every time captain runs, every captain request will make a connection to each of those services, when actually, they only need to make connections to the services that their corresponding current step is going to use.
+
+Letting you pass in factories is a handy way to avoid this, since when calling `create` you would just initialize the factories and not the underlying object. But still, depending on how you initialize submitters/status retrievers/ handle persistors using factories might not be necessary. Especially if you're trying to share connections across waypoints or manifests, you may push the depedency up your hierarchy in which case the factory is less likely to be valuable. Do what works for your infra.
+
+
+### Gotchas, Notes on the Iface, Miscellania
+
+- Captain is a busy loop. Be careful what you log. You can very quickly nuke your inbox or your disk.
+- Captain is a busy loop. Be thoughtful about interactions with external resources (e.g. db interactions).
+- Captain is a busy loop. It provides various controls to prevent it from spinning too fast (e.g. `setConfigWaitTime`). Most of the defaults selected for these controls are intended to provide safe usage. That being said, as you're implementing Captain, be mindful that the busy loop is "spinning" at a reasonable rate. 
+- parallelExecutions: num of requests a captain node should process in parallel. (5 - 20 appears to be pretty conventional. Ideally each time Captain "processes" (meaning any single waypoint) a request the amount of works its doing should take less than a second. So even 5 threads can move you through requests very quickly) 
+
+Did you know Captain is a busy loop?
+
+#### Emulating a DAG
```

Comment:
@h-wang94 - you may have strong opinions on this section as you're actually working on this stuff.

Reply:
Ya, i think this is fine for now.

Hoping to make changes ...sometime in the future that may make it easier for people who want linear workflows also.

Reply:
:+1: